jenkins/scripts/select-compiler.sh

```diff
@@ -143,6 +154,10 @@ elif [ "$SELECT_ARCH" = "X64" ]; then
   echo "Setting compiler for Node version $NODEJS_MAJOR_VERSION on x64"
 
   case $nodes in
+    centos7-x64 )
```

Comment:
This and the arm64 below aren't correct. It appears that https://ci.nodejs.org/job/node-test-commit-linux/ works differently from the other platforms above, which base the compiler decision on machine name (test/release-blah-blah_-x) + the nodejs version.

These ones take a different approach, its based on configuration name (`$nodes`), so a new configuration will need to be set for devtools-8 (probably centos7-gcc8). I'll have to work out how that relates to nodejs release version, if at all.

If anyone has some historic context and understands how this should work for devtools-8, feel free to chime in, otherwise I'll wing it.

Reply:
Historical context is probably that the ones using `$NODE_NAME` are the original form of this script which was limited to just the IBM platforms, so selection was a fairly straightforward affair and it ignored use of VersionSelectorScript, hence the liberal use of version selecting in here. The ones using `$nodes` are when we expanded it to more platforms and use labels instead to get broad selection and match how we're selecting nodes in our jobs and it uses VersionSelectorScript.

The use of labels lets us repurpose the same machine for different things and combine this script with VersionSelectorScript so we can switch its roles. Ideally `$NODEJS_MAJOR_VERSION` and anything like it should be removed from this script so version switching was isolated only to VersionSelectorScript, having it all in a single place (IMO!).

How you do this: assign the same machine(s) multiple labels indicating different compiler configurations (as in here where you have a `-gcc48` postfix. The job gets all those labels, then VersionSelectorScript tells the job which ones to select. By the time you get to this script, you have all the info you need for selecting the compiler contained within the label so you should just need to branch on that and not do anything with the Node.js version.

Reply:
OK, -gcc6 means devtools-6, I should keep the convention, use -gcc8 for devtools-8?

And for example, the VersionSelectorScript needs to change for 14.x to exclude centos7-x64-gcc6 for release (and test?), and allow centos7-x64-gcc8 for release (and test)?

Reply:
yeah, you should just be able to extend the centos block in VersionSelectorScript: https://github.com/nodejs/build/blob/050fb9733697acb3dd5bb53891c2818e2c335bab/jenkins/scripts/VersionSelectorScript.groovy#L24-L27
excluding centos6 from the optionality (`[67]`).

then in the centos7 machines, add a new label to each of them: https://ci.nodejs.org/label/centos7-64-gcc6/

then in linux-test-commit you can add that new label to the list:

<img width="154" alt="Screenshot 2020-04-07 15 54 22" src="https://user-images.githubusercontent.com/495647/78634802-1fd4b180-78e8-11ea-814e-00b4febb0b77.png">

There may also be some stuff inline in node-test-commit-linux that matches labels that will need tweaking, better check the script invocations in there.

If you want to tinker with this, copy node-test-commit-linux to a temporary one, point the select-compiler.sh run command to point to your fork/branch and do the same for VersionSelectorScript so it's pulling from a copy you're editing. Be aware that there's some caching that can get in the way when you're editing VersionSelectorScript live so you may need to add a query string on the end and change it each time you edit (`?1` etc.).

Reply:
and re gcc6 and gcc8 vs devtools-6 and devtools-8, it's not critical but probably better to use gcc version because this same scheme can apply outside of centos